src/core_plugins/vega/public/vega_view/vega_tooltip.js

```diff
@@ -0,0 +1,78 @@
+import { calculatePopoverPosition } from '@elastic/eui';
+import { formatValue as createTooltipContent, escapeHTML } from 'vega-tooltip';
+
+// Some of this code was adapted from https://github.com/vega/vega-tooltip
+
+const tooltipId = 'vega-kibana-tooltip';
+
+/**
+ * Simulate the result of the DOM's getBoundingClientRect()
+ */
+function createRect(left, top, width, height) {
+  return {
+    left, top, width, height,
+    x: left, y: top, right: left + width, bottom: top + height,
+  };
+}
+
+/**
+ * The tooltip handler class.
+ */
+export class TooltipHandler {
+  constructor(container, view, opts) {
+    this.container = container;
+    this.position = opts.position;
+    this.padding = opts.padding;
+    this.centerOnMark = opts.centerOnMark;
+
+    view.tooltip(this.handler.bind(this));
+  }
+
+  /**
+   * The handler function.
+   */
+  handler(view, event, item, value) {
+    this.hideTooltip();
+
+    // hide tooltip for null, undefined, or empty string values
+    if (value == null || value === '') {
+      return;
+    }
+
+    const el = document.createElement('div');
+    el.setAttribute('id', tooltipId);
+    el.classList.add('euiToolTipPopover', 'euiToolTip', `euiToolTip--${this.position}`);
+
+    // Sanitized HTML is created by the tooltip library,
+    // with a largue nmuber of tests, hence supressing eslint here.
+    // eslint-disable-next-line no-unsanitized/property
+    el.innerHTML = createTooltipContent(value, escapeHTML);
```

Comment:
Given the context of where the escaped HTML is being input, `vega-tooltip`'s `escapeHTML` function appears to be "safe enough". However, it's only escaping `<` and `&` so if `vega-tooltip` began using `value` to set attributes on HTML elements, we would be vulnerable to an XSS. We're using `_.escape` elsewhere to escape HTML, so i'd prefer we use `_.escape` instead of `vega-tooltip.escapeHTML` to further protect ourselves.